File: frontend/src/api/jobs.ts
import { apiGet, apiPost } from "./client";

export type JobStatus = "running" | "succeeded" | "failed" | "needs_review";

export interface Job {
  id: string;
  source_id: string;
  status: JobStatus;
  error: string | null;
  created_at: string;
  updated_at: string;
}

export function processSource(sourceId: string): Promise<Job> {
  return apiPost<Job>("/jobs/process", { source_id: sourceId });
}

export function fetchJob(jobId: string): Promise<Job> {
  return apiGet<Job>(`/jobs/${encodeURIComponent(jobId)}`);
}

File: backend/app/schemas/jobs.py
from datetime import datetime

from pydantic import BaseModel


class ProcessSourceRequest(BaseModel):
    source_id: str


class JobOut(BaseModel):
    id: str
    source_id: str
    status: str
    error: str | None = None
    created_at: datetime
    updated_at: datetime


class JobListResponse(BaseModel):
    jobs: list[JobOut]
